code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,197 @@
+/* 	< OH MY GOD. Can't you just make "/image/proc/foo()" instead of making these? >
+ * 		/appearance is a hardcoded byond type, and it is very internal type.
+ *		Its type is actually /image, but it isn't truly /image. We defined it as "/appearance"
+ * 		new procs to /image will only work to actual /image references, but...
+ * 		/appearance references are not capable of executing procs, because these are not real /image
+ * 		This is why these global procs exist. Welcome to the curse.
+ */
+/// Makes a var list of /appearance type actually uses. This will be only called once.
+/proc/build_virtual_appearance_vars()
+	. = list("vis_flags") // manual listing
+	var/list/unused_var_names = list(
+		"appearance", // it only does self-reference
+		"x","y","z", // these are always 0
+        "weak_reference", // it's not a good idea to make a weak_ref on this, and this won't have it
+        "vars", // inherited from /image, but /appearance hasn't this
+
+		// we have no reason to show those, right?
+		"active_timers",
+		"comp_lookup",
+		"datum_components",
+		"signal_procs",
+		"status_traits",
+		"gc_destroyed",
+		"stat_tabs",
+		"cooldowns",
+		"datum_flags",
+		"visibility",
+		"verbs",
+		"tgui_shared_states"
+		)
+	var/image/dummy_image = image(null, null)
+	for(var/each in dummy_image.vars) // try to inherit var list from /image
+		if(each in unused_var_names)
+			continue
+		. += each
+	del(dummy_image)
+	dummy_image = null
+
+/// appearance type needs a manual var referencing because it doesn't have "vars" variable internally.
+/// There's no way doing this in a fancier way.
+/proc/debug_variable_appearance(var_name, image/appearance)
```

Comment:
If the thing coming in isn't an image, then it shouldn't be typed as an image

Reply:
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/87972842/2943e426-52db-454e-b2a7-5e8064fa0ede)

Welcome to Byond

Reply:
At least it's a single case. I converted it to movable.

Reply:
thats even worse, its definitely not an atom/movable. Its a variable type and should be untyped, with the type being resolved later on.

Reply:
aight changed